rtc/encryption.md

```diff
@@ -0,0 +1,70 @@
+
+> **Note**: {{site.companyname}} is in the process of changing RTC encryption to [AES-KW](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/wrapKey) which does not require an initialization vector or protocol salt. The description below is still correct for the current beta release.
+
+The RTC protocol encryption technique is as follows:
+* 256 bits of random data are generated as the salt for each session. As described above, each document ID used for collaboration may have multiple sessions. The salt is generated with the [Crypto.getRandomValues()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues) browser API and a `Uint8Array` of length 32 bytes.
+* The salt data and the provided encryption key are combined using the [PBKDF2](https://en.wikipedia.org/wiki/PBKDF2) key derivation function. Derivation is performed by the [SubtleCrypto.deriveKey()](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/deriveKey) browser API, using the following parameters:
+  * PBKDF2 [algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Pbkdf2Params), with SHA-512 digest algorithm and 10000 iterations
+  * AES-GCM [derivedKeyAlgorithm](https://developer.mozilla.org/en-US/docs/Web/API/AesKeyGenParams) with length 256
+* The derived key is cached (in memory only) and used with [SubtleCrypto.encrypt()](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt) and [SubtleCrypto.decrypt()](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/decrypt) as necessary, specifying the [AES-GCM algorithm](https://developer.mozilla.org/en-US/docs/Web/API/AesGcmParams). For each encrypted message 96 bits of random data are used as the initialization vector, also generated with [Crypto.getRandomValues()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues) but this time using `Uint8Array` length 12. Each IV is transmitted alongside the encrypted message.
+* The provided key and derived key are never transmitted between clients. Each client will already have the provided key, so only the salt needs to be transmitted for clients to recreate the derived key. The salt alone cannot be used to decrypt the document contents.
+* When a session is established the provided `keyHint` and protocol-generated salt are sent to the server via a HTTPS websocket connection where they are stored in a database entry linked to the session. When subsequent clients connect to the session, the server sends these values along with other session information so the client can replicate the unique session key using [rtc_encryption_provider]({{site.baseurl}}/rtc/configuration#rtc_encryption_provider) and key derivation.
```

Comment:
```suggestion
* When a Real-time collaboration session is established, the provided `keyHint` and protocol-generated salt are sent to the server through a HTTPS websocket connection, where they are stored in a database-entry linked to the session. When subsequent clients connect to the session, the server sends these values along with other session information so the client can replicate the unique session key using [rtc_encryption_provider]({{site.baseurl}}/rtc/configuration#rtc_encryption_provider) and key derivation.
```

Reply:
`third-party` doesn't have a hyphen, but `database-entry` does? 🧐

Reply:
no, it does. if I missed it, please fix third-party refs (hyphen required)
Added the hyphen here for readability. those three words in a row are hard to parse, so the hyphen was added to show association between to the two most-related words.

Reply:
Ah my apologies I misremembered the earlier change - you _added_ one, you didn't remove one. There's a lot to keep track of.